### DISCUSSION TITLE
Thoughts on Online Web Based DAW development

### DISCUSSION THREAD

**Discussion by Meesam-12321** [LMMS/lmms#7494] [General]
Hi guys! I am planning to develop an online DAW, since you guys are aware that most of such products are desktop based applications and my peers are raising questions on its feasibility on the web, because most of the plugins are written using c++ and there can be performance bottlenecks for a MERN stack based DAW, What do you think, should I give it a go?? Actually, if it was a hobby project then i wouldnt have thought much and just started without having feasibility concerns but since it is gonna be my FYP, I cant take chances. Because if I get stuck in an issue in the middle of development , then it would be wastage of alot of time and starting again would make it harder for me to meet the deadlines. What techstack do you guys propose?

**Comment by Rossmaxx** [comment]
Web is inherently slower for audio applications and afaik, realtime and web never goes hand in hand. There's this thing with audio that every sample should be processed 44100 times every second, so that might introduce lotta lag. There is an online daw named trackton waveform that i know of, but don't know the inner workings tho. 

Anyway give it a shot. You'll learn a lot, even if the effort ultimately gets wasted, the learnings won't. 

If you get stuck, feel free to ask here for help. Or a better place would be the discord server. It's easier to have a chat over there.

**Comment by ghost** [comment]
cant we just use bandlab?

**Comment by softrabbit** [comment]
Shouldn't be impossible to make some kind of DAW using this:
https://webaudio.github.io/web-audio-api/#processing-model-background

**Comment by Cybernetic-Fox** [comment]
This is interesting! I'm not quite keen on browser DAWS, since (like Rossmaxx says, and my experiences with web-based DAWS) they can unintentionally introduce a huge buffer that heavily limits playback. 

A tendency with browsers such as Google Chrome is that the processes within the web application can cause serious buffers depending on the _user's_ device, introducing even more buffer/lag that minimizes capabilities and functions on large-scale projects.

I think you should strive and attempt to develop such a project because you may know how to combat this or learn how to improve what other web-based DAWS struggle with.

Softrabbit's source gives quite a unique view of processing!

This is only my thoughts based on my experiences :D

**Reply by Meesam-12321** [reply]
Thankyou very much for such a motivating reply. It is really a green signal for me to get started and when I actually become successful in the development of this product, this is gonna be the first place I am gonna share the outcomes. Again thanks!!!

**Reply by Cybernetic-Fox** [reply]
Of course! You're so welcome! :)

**Reply by SussyNavillus** [reply]
How long do you think this will take based on what you have rn.

**Reply by Meesam-12321** [reply]
I am just setting up things right now and learning through trial and error figuring out which approach would be the best. I came across web assembly and learned how it provides native level performance on the web. I will try to create some demo plugins and once they work fine. I will put my everything into it.

**Reply by SussyNavillus** [reply]
Did you get it working or are you still fixing problems

**Comment by sportzfyy** [comment]
Online web-based Digital Audio Workstations (DAWs) offer numerous advantages and challenges for both developers and users. They provide accessibility, collaboration, and ease of use across different platforms, with no installation required. They offer built-in collaboration tools, cost-effectiveness, and cloud integration, making them accessible for beginners and casual musicians. Projects are saved directly to the cloud, allowing users to access their work from anywhere without manual backup. Web-based DAWs can be continuously updated without user intervention, providing users with the latest tools and capabilities. However, they face challenges such as latency and performance issues, limited plugin support, dependence on internet connection, limited offline functionality, and security and privacy concerns. Popular web-based DAWs include Soundation, BandLab, and Audiotool. As web technologies and cloud computing evolve, more powerful and feature-rich online DAWs are expected to emerge. The increasing demand for remote collaboration and cloud-based workflows make web-based DAWs highly relevant in modern music production. In conclusion, while online web-based DAWs offer accessibility, collaboration, and ease of use, desktop DAWs are still the preferred choice for users requiring advanced features, offline capabilities, and low-latency performance.

**Comment by sportzfyy** [comment]
An online web-based Digital Audio Workstation (DAW) offers numerous advantages, including accessibility, cross-platform support, real-time collaboration, automatic updates, and cost-effectiveness. However, challenges in development include audio latency, browser limitations, offline functionality, complexity, and security. Key technologies include the Web Audio API, WebAssembly, cloud storage integration, AI and Machine Learning, and AI and Machine Learning integration.

Use cases for web-based DAWs include beginning musicians, collaborative projects, education, and quick edits for professionals. A simplified interface for learning and experimenting is ideal for beginners, while collaborative projects allow teams to work remotely. Education solutions are ideal for classrooms or workshops, while quick edits allow professionals to make adjustments on the go without accessing their primary setup.

Popular web-based **[sportzfy](https://sportzfyy.net/)** like Soundation, BandLab, and Audiotool demonstrate the feasibility and growing interest in this space. To develop a well-designed web-based DAW, prioritize intuitive design, latency optimization, and robust collaborative features. With continuous improvements in web technologies, a well-designed web-based DAW could disrupt the industry by making music production more accessible than ever.

**Comment by Rossmaxx** [comment]
Why is this thread so attractive to bots?

**Comment by messmerd** [comment]
Locking this discussion due to bots
